modules/calc/src/main/java/com/opengamma/strata/calc/runner/DerivedCalculationFunctionWrapper.java

```diff
@@ -0,0 +1,194 @@
+   * <p>
+   * The inputs to the measure can include measures calculated by the delegate calculation function.
+   */
+  private final DerivedCalculationFunction<T, R> calculationFunction;
+
+  /**
+   * A calculation function whose results can be used by the derived calculation function.
+   */
+  private final CalculationFunction<T> delegate;
+
+  /**
+   * The measures supported by this function; the union of the measures supported by the delegate function and
+   * the derived function.
+   */
+  private final Set<Measure> supportedMeasures;
+
+  /**
+   * True if the delegate function supports all measures required by the calculation function.
+   * If this is true the calculation function can be invoked.
+   * If it is false the only measures which can be calculated are the measures supported by the delegate.
+   */
+  private final boolean requiredMeasuresSupported;
+
+  /**
+   * Creates a new function which invokes the delegate function, passes the result to the derived function
+   * and returns the combined results.
+   *
+   * @param calculationFunction  a function which calculates one measure using the measure values calculated by the other function
+   * @param delegate  a function which calculates multiple measures
+   */
+  DerivedCalculationFunctionWrapper(
+      DerivedCalculationFunction<T, R> calculationFunction,
+      CalculationFunction<T> delegate) {
+
+    this.calculationFunction = calculationFunction;
+    this.delegate = delegate;
+
+    Set<Measure> delegateMeasures = delegate.supportedMeasures();
+    this.requiredMeasuresSupported = delegateMeasures.containsAll(calculationFunction.requiredMeasures());
+    this.supportedMeasures = requiredMeasuresSupported ?
+        ImmutableSet.<Measure>builder().addAll(delegateMeasures).add(calculationFunction.measure()).build() :
+        delegateMeasures;
+  }
+
+  @Override
+  public Class<T> targetType() {
+    return calculationFunction.targetType();
+  }
+
+  @Override
+  public Set<Measure> supportedMeasures() {
+    return supportedMeasures;
+  }
+
+  @Override
+  public Currency naturalCurrency(T target, ReferenceData refData) {
+    return delegate.naturalCurrency(target, refData);
+  }
+
+  @Override
+  public FunctionRequirements requirements(
+      T target,
+      Set<Measure> measures,
+      CalculationParameters parameters,
+      ReferenceData refData) {
+
+    FunctionRequirements delegateRequirements = delegate.requirements(target, measures, parameters, refData);
+    FunctionRequirements functionRequirements = calculationFunction.requirements(target, parameters, refData);
+    return delegateRequirements.combinedWith(functionRequirements);
+  }
+
+  @Override
+  public Map<Measure, Result<?>> calculate(
+      T target,
+      Set<Measure> measures,
+      CalculationParameters parameters,
+      ScenarioMarketData marketData,
+      ReferenceData refData) {
+
+    // Does the caller want the extra measure?
```

Comment:
This code is a little tricky to follow even with comments. Would it read more clearly as

if (measureRequested) // handle and return
if (!requiredMeasuresSupported)  // handle and retrun
//handle required and supported